client/mapView/MapView.cpp

```diff
@@ -115,9 +123,37 @@ void MapView::onMapScrolled(const Point & distance)
 
 void MapView::onMapSwiped(const Point & viewPosition)
 {
+	swipeHistory[GH.input().getTicks()] = viewPosition;
+
 	controller->setViewCenter(model->getMapViewCenter() + viewPosition, model->getLevel());
 }
 
+void MapView::postSwipe(uint32_t msPassed) {
+	if(!actions->dragActive)
+	{
+		if(swipeHistory.size() > 1)
+		{
+			Point diff = Point(0, 0);
+			for (auto & x : swipeHistory)
+				diff += x.second;
+
+			uint32_t timediff = swipeHistory.rbegin()->first - swipeHistory.begin()->first;
+
+			postSwipeAngle = diff.angle();

```

Comment:
what if both diff.x and diff.y are zeros? Maybe good to check this case explicitly

Reply:
Results in 0. And in this case the speed is also 0. So no problem in this case.